index.js

```diff
@@ -1,18 +1,182 @@
+    answerYesNo = await ask("\nIs it ... " + guess + "? (Y/N) ")
+
+    // ------------- loop for computer's guesses -------------------------------------------------------------------------- 
+
+    while (!yesAnswer.includes(answerYesNo.toLowerCase())) {
+      answerHighLow = await ask("\nIs it higher (H) or lower (L)? ")
+
+      // -------------------- guessing mechanism --------------------------------------------------------------------------
+
+      if (highAnswer.includes(answerHighLow.toLowerCase())) {
+        min = guess + 1;
+
+      } else if (lowAnswer.includes(answerHighLow.toLowerCase())) {
+        max = guess - 1;
+      } else {
+        answerHighLow = await ask("\nIs it higher (H) or lower (L)? ")
+      }
+      guess = binarySearch(min, max)
+      answerYesNo = await ask("\nIs it ... " + guess + "? (Y/N) ")
+      count += 1;
+
+      // --------------- cheat detector checks if player gave inaccurate responses ------------------------------------------
+
+      if (min > max) {
+        console.log("\nThere appears to be a problem. Are you sure about your number?");
+        process.exit();
+      }
+    }
+
+    // ------------------ conclusion to game and offer to play the other game ----------------------------------------------
+
+    console.log("\nWow! I guessed it in " + count + " tries!")
+    let playOtherGame = await ask("\nNow would you like to guess my number? Y/N ")
+    if (yesAnswer.includes(playOtherGame.toLowerCase())) {
+      playerGuess()
+    } else {
+      console.log("\nThanks for playing!")
+      process.exit();
+    }
+  }
+
+  // ---------------------------------------------------------------------------------------------------------------------
+  // --------- interactive game where player guesses computer's randomly generated number --------------------------------
+
+  async function playerGuess() {
+
+    // --------- checks that the range numbers input are actually numbers ---------------------------------------------------
+
+    let lowerRange = parseInt(await ask("\nPlease give me a minimum value. "));
+    let upperRange = parseInt(await ask("\nPlease give me a maximum value. "))
+    let i = 1
+
+    while (isNaN(lowerRange) || isNaN(upperRange)) {
+      if (isNaN(lowerRange)) {
+        lowerRange = parseInt(await ask("\nPlease give me a valid minimum value. "));
+      }
+      if (isNaN(upperRange)) {
+        upperRange = parseInt(await ask("\nPlease give me a valid maximum value. "))
+      }
+      if (i === 2) {
+        process.exit()
+      }
+      i++
+    }
+
+    // ----------- computer generates a random number using the specified range --------------------------------------------
+
+    let compNum = randomInt(upperRange, lowerRange);
+    let playerGuess = parseInt(await ask("\nGuess a number. "))
+    let count = 1;
+
+    // ------------- loop for player's guesses ---------------------------------------------------------------------------
```

Comment:
Good comments throughout your code to separate out sections of your code and to describe different blocks of logic. Comments that separate out different sections like your functions, your game code, global variables, etc are okay to use dashes across the file, but areas describing blocks of logic like if-else branches and while loops a regular comment will suffice.
```js
// ------------ this is a comment separating out sections -----------------
function start() {
  // this is a comment describing my while loop 
  while (true) {
    // do some stuff
  }
}

```